source/Engine.cpp

```diff
@@ -2080,14 +2080,14 @@ void Engine::DoCollisions(Projectile &projectile)
 
 				// Only directly targeted ships get provoked by blast weapons.
 				int eventType = ship->TakeDamage(visuals, damage.CalculateDamage(*ship, ship == hit.get()),
-					targeted ? gov : nullptr);
+					targeted ? gov : nullptr, projectile.Position() + projectile.Velocity() * closestHit, hitVelocity, projectile.Facing());
 				if(eventType)
 					eventQueue.emplace_back(gov, ship->shared_from_this(), eventType);
 			}
 		}
 		else if(hit)
 		{
-			int eventType = hit->TakeDamage(visuals, damage.CalculateDamage(*hit), gov);
+			int eventType = hit->TakeDamage(visuals, damage.CalculateDamage(*hit), gov, projectile.Position() + projectile.Velocity() * closestHit, hitVelocity, projectile.Facing());
```

Comment:
How about a linebreak here?

Reply:
I was trying to keep the style of the code the same (didn't have a linebreak previously), but given the length of the statement it may be worth it for human readability.

Reply:
The rule is to keep the length of lines below 100 (or 120, not sure)

Reply:
Those lines in Engine.cpp have been sorted out now.